### PR TITLE
Improve jobs and scheduler

### PR DESCRIPTION
**Pull request by gabrielpoca**
In this change:

* Introduce the `Job` entity to keep track of job execution.
* Extract the logger into a separate module that can be used across the app.
* Create a decorator function to control job execution.
* Upgrade bull to handle memory leak.
* Extract API to an `api` folder.

The idea, later on, is to extract the scheduler into a separate node process that runs in parallel with the API. The idea with running the jobs every hour (for now) is to ensure that we still run a job if the server is down.

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import logger from '../logger';
+
+export const server = fastify({
+  logger,
+});
+
+server.get('/ping', async (_request, _reply) => {
+  return 'pong\n';
+});
```
Comment (joaop21) [inline src/api/index.ts:11]: Is it ok to have the routes definitions here? Or will they eventually be moved somewhere?

**Review by joaop21 — APPROVED** [review]
👍  🚀

File: src/api/index.ts (hunk 1 above)
Comment (gabrielpoca) [reply src/api/index.ts:11]: I was just moving API related code to the `api` folder. I guess it's fine to have everything in the same file if we only have a few.

**Comment by 0xrin1** [comment]
cron setting string could use adjusting so it doesnt run every minute but to be honest even every minute would be OK since this load doesn't scale with user number

LGTM :muscle:

**Review by 0xrin1 — APPROVED** [review]

**Pull request merged**